test/meta_test.js

```diff
@@ -42,6 +42,17 @@ suite('Repo Meta Tests', function() {
     assert.equal(taskclusterYml.tasks.$let.node, uiPackageJson.engines.node);
   });
 
+  test('git grep test', async function() {
+    const Taskcluster = ["Task[C]luster", "Task [c]luster", "Task [C]luster"]
+    Taskcluster.forEach( value => {
+      exec(`git grep ${value}`).then((res)=>{
+        console.log(res)
```

Comment:
This will echo something to stdout, but does not cause the test to fail:
```
  Repo Meta Tests
    ✓ All packages in CI (381ms)
    ✓ Node version in .taskcluster.yml matches that in package.json
    ✓ Node version for UI matches the rest of the repo
    ✓ git grep test
{ stdout: 'README.md:TaskCluster\n', stderr: '' }
    ✓ Dependencies are not missing/unused (6697ms)


  5 passing (7s)
```

Also, I think I see why you were getting unhandled errors -- nothing is waiting for the `exec` Promise, so mocha carries on with the next test case immediately.  If you were to add an `assert` here, and it failed you would get an unhandled exception:
```
  Repo Meta Tests
    ✓ All packages in CI (376ms)
    ✓ Node version in .taskcluster.yml matches that in package.json
    ✓ Node version for UI matches the rest of the repo
    ✓ git grep test
{ stdout: 'README.md:TaskCluster\n', stderr: '' }
(node:14947) UnhandledPromiseRejectionWarning: AssertionError [ERR_ASSERTION]: Input A expected to strictly equal input B:
+ expected - actual

- undefined
+ 1
    at exec.then.catch (/home/dustin/p/taskcluster/test/meta_test.js:52:16)
(node:14947) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:14947) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

The easiest way to fix this is to use a `for .. of` loop, which allows you to use `await` in the body of the loop:
```js
for (let pattern of Taskcluster) {
  try {
    res = await exec(`git grep '${pattern}'`); // <-- note quoting the pattern for the shell
    // if the grep succeeded, then something matched
    throw new Error(`misspellings found: ${res.stdout}`);
  } catch (err) {
    if (err.code === 1) {
      // git grep found nothing
      return;
    }
    throw err;
  }
}
```

As a general rule, it's best to avoid mixing async/await (and you've used `async` to define this test function) with `.then/.catch`.  And async/await work better with `for` loops than with array methods like `foreach` and `map`.